Source/RunActivity/Viewer3D/Lights.cs

```diff
@@ -102,6 +103,8 @@ public LightViewer(Viewer viewer, TrainCar car, TrainCarViewer carViewer)
                     {
                         case LightType.Glow:
                             LightPrimitives.Add(new LightGlowPrimitive(this, Viewer.RenderProcess, light));
+                            if (light.Graphic != null)
+                                (LightPrimitives.Last() as LightGlowPrimitive).SpecificGlowMaterial = viewer.MaterialManager.Load("LightGlow", DefineFullTexturePath(light.Graphic));
```

Comment:
Some other parts of the code could be simplified if `SpecificGlowMaterial` was set equal to the `LightGlowMaterial` if `light.Graphic` is null.

On line 235 and 237, a null check is performed every frame to select between the specific glow material or the general glow material, but that check could effectively be done one time on initialization by changing this.
```suggestion
                            if (light.Graphic != null)
                                (LightPrimitives.Last() as LightGlowPrimitive).SpecificGlowMaterial = viewer.MaterialManager.Load("LightGlow", DefineFullTexturePath(light.Graphic));
                            else
                                (LightPrimitives.Last() as LightGlowPrimitive).SpecificGlowMaterial = LightGlowMaterial;
```